src/WebJobs.Extensions.DurableTask/Listener/TaskHttpActivityShim.cs

```diff
@@ -42,32 +42,40 @@ public async override Task<string> RunAsync(TaskContext context, string rawInput
             HttpRequestMessage requestMessage = await this.ConvertToHttpRequestMessage(durableHttpRequest);
 
             HttpResponseMessage response;
-            if (durableHttpRequest.Timeout == null)
+            try
             {
-                response = await this.httpClient.SendAsync(requestMessage);
-            }
-            else
-            {
-                try
+                if (durableHttpRequest.Timeout == null)
                 {
-                    using (CancellationTokenSource cts = new CancellationTokenSource())
-                    {
-                        cts.CancelAfter(durableHttpRequest.Timeout.Value);
-                        response = await this.httpClient.SendAsync(requestMessage, cts.Token);
-                    }
+                    response = await this.httpClient.SendAsync(requestMessage);
                 }
-                catch (OperationCanceledException ex)
+                else
                 {
-                    TimeoutException e = new TimeoutException(ex.Message + $" Reached user specified timeout: {durableHttpRequest.Timeout.Value}.");
+                    try
+                    {
+                        using (CancellationTokenSource cts = new CancellationTokenSource())
+                        {
+                            cts.CancelAfter(durableHttpRequest.Timeout.Value);
+                            response = await this.httpClient.SendAsync(requestMessage, cts.Token);
+                        }
+                    }
+                    catch (OperationCanceledException ex)
```

Comment:
Can we just do the `catch (HttpRequestException)` here at the same level as `catch (OperationCanceledException)` instead of putting it into a new try/catch block?

Reply:
The `catch (OperationCanceledException)` is specifically for the case where `durableHttpRequest.Timeout` is set and the `catch (HttpRequestException)` is applicable to whether or not the Timeout property is set. The `catch(OperationCanceledException)` could move one level out to be consistent with `catch (HttpRequestException)`.

Reply:
I'm just wondering why you couldn't put `catch (HttpRequestException)` on line 68. That would be a smaller change and I don't think that exception can be thrown anywhere else.

Reply:
I don't think that would cover the `await this.httpClient.SendAsync(requestMessage)` call on line 49 if it throws an HttpRequestException.

Reply:
Oh! I totally missed that other send on line 49. What you have makes sense then. However, it makes me wonder if we should consider also handing `OperationCanceledException` in the outer catch block as well. We might not handle it exactly the same since there isn't a user-specified timeout in the case of line 49, but it might be a good idea so that there is a consistent experience for HTTP timeouts, whether it's the default timeout or a user-provided timeout.